akka-remote/src/main/scala/akka/remote/Endpoint.scala

```diff
@@ -1035,13 +1035,12 @@ private[remote] class EndpointReader(
       replyTo ! StoppedReading(writer)
 
     case InboundPayload(p) if p.size <= transport.maximumPayloadBytes ⇒
-      val (ackOption, msgOption) = tryDecodeMessageAndAck(p)
-      for (ack ← ackOption; reliableDelivery ← reliableDeliverySupervisor) reliableDelivery ! ack
-
-      if (log.isWarningEnabled)
+      if (log.isWarningEnabled) {
+        val (_, msgOption) = tryDecodeMessageAndAck(p)
```

Comment:
does `tryDecodeMessageAndAck` update the `AckedReceiveBuffer`? then that might have to be adjusted so that it doesn't think that it has received the message and deduplicates it when it arrives again. It should only tryDecode

Reply:
Doesn't look like it, only updated in `receive` not in `notReading`

Reply:
ok, good